lang/src/gen/as_dependency.rs

```diff
@@ -237,6 +238,11 @@ fn generate_call_enhancer_messages<'a>(
             let output = &message.sig.decl.output;
             let (_impl_generics, type_generics, where_clause) = message.sig.decl.generics.split_for_impl();
             let selector = message.selector();
+            let selector = format!(
+                "[{}, {}, {}, {}]",
+                selector[0], selector[1], selector[2], selector[3]
+            );
+            let selector = parse_str::<syn::Expr>(&selector).expect("failed to parse selector");
```

Comment:
same question here. i see a potential code dupe. maybe we want to pull that code out and do this just once.